BaikalDataGenerator/Source/render.cpp

```diff
@@ -0,0 +1,351 @@
+                image_data[y * width + x] = val;
+            }
+        }
+    }
+
+    std::stringstream ss;
+
+    ss << "cam_" << cam_index << "_"
+        << desc.name << "_spp_" << spp << ".png";
+
+    std::filesystem::path file_name = output_dir;
+    file_name.append(ss.str());
+
+    std::unique_ptr<ImageOutput> out(ImageOutput::create(ss.str()));
+
+    if (!out)
+    {
+        THROW_EX("Can't create image file on disk");
+    }
+
+    ImageSpec spec(width, height, 3, TypeDesc::FLOAT);
+    out->open(file_name.string(), spec);
+    out->write_image(TypeDesc::FLOAT, image_data.data(), sizeof(float3));
+    out->close();
+}
+
+void Render::SetLight(const std::vector<LightInfo>& light_settings)
+{
+    // clear light
+    auto iter = m_scene->CreateLightIterator();
+    while (iter->IsValid())
+    {
+        m_scene->DetachLight(iter->ItemAs<Baikal::Light>());
+        iter = m_scene->CreateLightIterator();
+    }
+
+    // set light
+    for (const auto& light: light_settings)
+    {
+        Light::Ptr light_instance;
+
+        if (light.type == "point")
+        {
+            light_instance = PointLight::Create();
+        }
+        else if (light.type == "direct")
+        {
+            light_instance = DirectionalLight::Create();
+        }
+        else if (light.type == "spot")
+        {
+            light_instance = SpotLight::Create();
+            SpotLight::Ptr spot = std::dynamic_pointer_cast<SpotLight>(light_instance);
+            spot->SetConeShape(light.cs);
+        }
+        else if (light.type == "ibl")
+        {
+            light_instance = ImageBasedLight::Create();
+
+            ImageBasedLight::Ptr ibl = std::dynamic_pointer_cast<
+                ImageBasedLight>(light_instance);
+
+            auto image_io(ImageIo::CreateImageIo());
+            // check that texture file is exist
+            auto texure_path = std::filesystem::path(light.texture);
+
+            if (!std::filesystem::exists(texure_path))
+            {
+                THROW_EX("textrue image doesn't exist on specidied path")
+            }
+
+            Texture::Ptr tex = image_io->LoadImage(light.texture);
+            ibl->SetTexture(tex);
+            ibl->SetMultiplier(light.mul);
+        }
+        else
+        {
+            THROW_EX("unsupported light type")
+        }
+
+        light_instance->SetPosition(light.pos);
+        light_instance->SetPosition(light.dir);
+        light_instance->SetPosition(light.rad);
+        m_scene->AttachLight(light_instance);
+    }
+}
+
+void Render::GenerateDataset(const std::vector<CameraInfo>& camera_states,
+                             const std::vector<LightInfo>& light_settings,
+                             const std::vector<int>& spp,
+                             const std::filesystem::path& output_dir,
+                             bool gamma_corection_enabled)
+{
+    using namespace RadeonRays;
+
+    if (!std::filesystem::is_directory(output_dir))
```

Comment:
There is the corresponding  check in the ValidateConfig function

Reply:
But its different entities